CMakeLists.txt

```diff
@@ -135,6 +136,103 @@ if (NOT (Boost_THREAD_LIBRARY STREQUAL monero_Boost_THREAD_LIBRARY_RELEASE))
   message(FATAL_ERROR "Boost libraries for monero build differs from this project")
 endif()
 
+# hidapi
+# https://github.com/moneroexamples/openmonero/blob/18370af8fa51a0e9d05dfe678dcd5bcb7fad30f7/cmake/FindHIDAPI.cmake#L26-L60
+find_library(HIDAPI_LIBRARY
+  NAMES hidapi hidapi-libusb)
+
+find_path(HIDAPI_INCLUDE_DIR
+  NAMES hidapi.h
+  PATH_SUFFIXES
+  hidapi)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(HIDAPI
+  DEFAULT_MSG
+  HIDAPI_LIBRARY
+  HIDAPI_INCLUDE_DIR)
+
+if(HIDAPI_FOUND)
+  set(HIDAPI_LIBRARIES "${HIDAPI_LIBRARY}")
+  if((STATIC AND UNIX AND NOT APPLE) OR (DEPENDS AND CMAKE_SYSTEM_NAME STREQUAL "Linux"))
+    find_library(LIBUSB-1.0_LIBRARY usb-1.0)
+    find_library(LIBUDEV_LIBRARY udev)
+    if(LIBUSB-1.0_LIBRARY)
+      set(HIDAPI_LIBRARIES "${HIDAPI_LIBRARIES};${LIBUSB-1.0_LIBRARY}")
+      if(LIBUDEV_LIBRARY)
+        set(HIDAPI_LIBRARIES "${HIDAPI_LIBRARIES};${LIBUDEV_LIBRARY}")
+      else()
+        message(WARNING "libudev library not found, binaries may fail to link.")
+      endif()
+    else()
+      message(WARNING "libusb-1.0 library not found, binaries may fail to link.")
+    endif()
+  endif()
+
+  set(HIDAPI_INCLUDE_DIRS "${HIDAPI_INCLUDE_DIR}")
+endif()
+
+mark_as_advanced(HIDAPI_INCLUDE_DIR HIDAPI_LIBRARY)
+
+if (NOT (HIDAPI_INCLUDE_DIR STREQUAL monero_HIDAPI_INCLUDE_DIR))
+  message(STATUS "Found HIDAPI_INCLUDE_DIR: ${HIDAPI_INCLUDE_DIR}")
+  message(STATUS "Found monero_HIDAPI_INCLUDE_DIR: ${monero_HIDAPI_INCLUDE_DIR}")
+  message(FATAL_ERROR "hidapi libraries for monero build differs from this project")
+endif()
+
+# openssl
+if (STATIC)
+  set(OPENSSL_USE_STATIC_LIBS TRUE)
+endif()
+find_package(OpenSSL REQUIRED)
+
+if (NOT (OPENSSL_CRYPTO_LIBRARY STREQUAL monero_OPENSSL_CRYPTO_LIBRARY))
+  message(STATUS "Found OPENSSL_CRYPTO_LIBRARY: ${OPENSSL_CRYPTO_LIBRARY}")
+  message(STATUS "Found monero_OPENSSL_CRYPTO_LIBRARY: ${monero_OPENSSL_CRYPTO_LIBRARY}")
+  message(FATAL_ERROR "openssl libraries for monero build differs from this project")
+endif()
+
+# zmq
+# https://github.com/monero-project/monero/blob/9aab19f349433687c7aaf2c1cbc5751e5912c0aa/CMakeLists.txt#L1171-L1199
```

Comment:
zmq changes in this file are here to fix compiler errors like:

```
/usr/bin/ld: /monero/build/release/src/net/libnet.a(zmq.cpp.o): in function `net::zmq::error_category()::category::message(int) const':
zmq.cpp:(.text+0xba): undefined reference to `zmq_strerror'
/usr/bin/ld: /monero/build/release/src/net/libnet.a(zmq.cpp.o): in function `net::zmq::terminate::call(void*)':
zmq.cpp:(.text+0x251): undefined reference to `zmq_errno'
/usr/bin/ld: zmq.cpp:(.text+0x25e): undefined reference to `zmq_term'
/usr/bin/ld: /monero/build/release/src/net/libnet.a(zmq.cpp.o): in function `net::zmq::receive[abi:cxx11](void*, int)':
zmq.cpp:(.text+0x2e4): undefined reference to `zmq_msg_init'
/usr/bin/ld: zmq.cpp:(.text+0x2f1): undefined reference to `zmq_msg_size'
/usr/bin/ld: zmq.cpp:(.text+0x2fe): undefined reference to `zmq_msg_data'
/usr/bin/ld: zmq.cpp:(.text+0x329): undefined reference to `zmq_msg_more'
/usr/bin/ld: zmq.cpp:(.text+0x33f): undefined reference to `zmq_msg_recv'
/usr/bin/ld: zmq.cpp:(.text+0x34b): undefined reference to `zmq_msg_close'
/usr/bin/ld: zmq.cpp:(.text+0x350): undefined reference to `zmq_errno'
/usr/bin/ld: zmq.cpp:(.text+0x3cc): undefined reference to `zmq_msg_close'
```

Pretty much just copy pasted the `monerod` code linked above.